cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/replication/ReplicationPolicyManager.java

```diff
@@ -0,0 +1,119 @@
+package com.cloudant.sync.replication;
+
+import com.cloudant.sync.notifications.ReplicationCompleted;
+import com.cloudant.sync.notifications.ReplicationErrored;
+import com.google.common.eventbus.Subscribe;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+public abstract class ReplicationPolicyManager {
+
+    private List<Replicator> replicators;
+    private ReplicationListener replicationListener;
+    private ReplicationsCompletedListener mReplicationsCompletedListener;
+
+    public interface ReplicationsCompletedListener {
+        void allReplicationsCompleted();
+
+        void replicationCompleted(int id);
+
+        void replicationErrored(int id);
+    }
+
+    private class ReplicationListener {
+
+        Set<Replicator> replicatorsInProgress;
+
+        ReplicationListener() {
+            replicatorsInProgress = Collections.synchronizedSet(new HashSet<Replicator>());
+        }
+
+        void add(Replicator replicator) {
+            replicatorsInProgress.add(replicator);
+        }
+
+        void remove(Replicator replicator) {
+            if (replicatorsInProgress.remove(replicator)) {
+                replicator.getEventBus().unregister(this);
+            }
+        }
+
+        boolean inProgress(Replicator replicator) {
+            return replicatorsInProgress.contains(replicator);
+        }
+
+        @Subscribe
+        public void complete(ReplicationCompleted event) {
+            finishedReplication(event.replicator);
+            if (mReplicationsCompletedListener != null) {
+                mReplicationsCompletedListener.replicationCompleted(event.replicator.getId());
+            }
+        }
+
+        @Subscribe
+        public void error(ReplicationErrored event) {
+            finishedReplication(event.replicator);
+            if (mReplicationsCompletedListener != null) {
+                mReplicationsCompletedListener.replicationErrored(event.replicator.getId());
+            }
+        }
+
+        public void finishedReplication(Replicator replicator) {
+            synchronized (replicatorsInProgress) {
+                remove(replicator);
+                if (replicatorsInProgress.size() == 0 && mReplicationsCompletedListener != null) {
+                    mReplicationsCompletedListener.allReplicationsCompleted();
+                }
+            }
+        }
+    }
+
+    public ReplicationPolicyManager() {
+        replicators = new ArrayList<Replicator>();
+        replicationListener = new ReplicationListener();
+    }
+
+    public abstract void start();
```

Comment:
Abstract methods `start()` and `stop()` now only serve to enforce an unnecessary naming convention on how to start/stop the policy. These should be removed and the class made concrete.